dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/UserAccountControllerTest.java

```diff
@@ -411,15 +412,15 @@ void selfRegInvalidRecaptchaInput() {
     systemSettingManager.saveSystemSetting(
         SettingKey.SELF_REGISTRATION_NO_RECAPTCHA, Boolean.FALSE);
 
-    assertWebMessage(
-        "Bad Request",
-        400,
-        "ERROR",
-        "Recaptcha validation failed: [invalid-input-secret]",
+    JsonWebMessage response =
         POST(
                 "/auth/registration",
                 renderService.toJsonAsString(getRegParamsWithRecaptcha("secret", RegType.SELF_REG)))
-            .content(HttpStatus.BAD_REQUEST));
+            .content(HttpStatus.BAD_REQUEST)
+            .as(JsonWebMessage.class);
+
+    assertTrue(response.getMessage().contains("Recaptcha validation failed"));
```

Comment:
We have https://github.com/dhis2/dhis2-core/blob/92f7623624bb1e695318f139e73578fd4ac51707/dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/test/utils/Assertions.java#L180 (or assertStartsWith) which give a better assertion message 😄 see https://github.com/dhis2/wow-backend/blob/master/guides/testing/testing.md#assertions

Reply:
cool, that's much nicer, I'll use that from now on 👍